### PR TITLE
Add PyPy finder

### PR DESCRIPTION
**Pull request by j178**
## Summary

This PR adds PyPy finder and adds PyPy to uv managed Python versions.

## Test Plan

```console
$ cargo run -- python install
```

### CODE REVIEW

File: .python-versions [hunk 1]
```diff
@@ -4,3 +4,5 @@
 3.9.18
 3.8.18
 3.8.12
+pypy@3.10.14
+pypy@3.9.19
```
Comment (zanieb) [inline .python-versions:8]: Maybe just use `uv python install` for these in the PyPy integration jobs we have in CI instead? Are you going to make use of these in unit tests?

Comment (j178) [reply .python-versions]: Thanks for pointing out, I wasn't aware of that.

File: docs/python-versions.md [hunk 2]
```diff
@@ -18,14 +18,12 @@ Many Python version request formats are supported:
...
-- `<implementation>` e.g. `cpython` or `cp`
-- `<implementation>@<version>` e.g. `cpython@3.12`
-- `<implementation><version>` e.g. `cpython3.12` or `cp312`
+- `<implementation>` e.g. `cpython` or `cp`, `pypy` or `pp`
+- `<implementation>@<version>` e.g. `cpython@3.12` or `pypy@3.12`
+- `<implementation><version>` e.g. `cpython3.12` or `cp312`, `pypy3.12` or `pp312`
 - `<implementation><version-specifier>` e.g. `cpython>=3.12,<3.13`
 - `<implementation>-<version>-<os>-<arch>-<libc>` e.g. `cpython-3.12.3-macos-aarch64-none`
 
-At this time, only CPython downloads are supported. However, PyPy support is planned.
```
Comment (zanieb) [inline docs/python-versions.md:27]: I would note something like "uv supports managed CPython and PyPy versions"

Comment (zanieb) [reply docs/python-versions.md:27]: Maybe we should note this elsewhere...

File: docs/python-versions.md [hunk 3]
```diff
@@ -18,14 +18,12 @@ Many Python version request formats are supported:
 
 - `<version>` e.g. `3`, `3.12`, `3.12.3`
 - `<version-specifier>` e.g. `>=3.12,<3.13`
-- `<implementation>` e.g. `cpython` or `cp`
-- `<implementation>@<version>` e.g. `cpython@3.12`
-- `<implementation><version>` e.g. `cpython3.12` or `cp312`
+- `<implementation>` e.g. `cpython` or `cp`, `pypy` or `pp`
+- `<implementation>@<version>` e.g. `cpython@3.12` or `pypy@3.12`
+- `<implementation><version>` e.g. `cpython3.12` or `cp312`, `pypy3.12` or `pp312`
```
Comment (zanieb) [inline docs/python-versions.md:23]: I wonder if we should be enumerating these all here? Maybe we should start a separate section for supported implementations — this is also missing GraalPy and I don't want to to be too verbose.

File: docs/python-versions.md (hunk 2 above)
Comment (j178) [reply docs/python-versions.md:27]: I added `uv bundles a list of downloadable CPython and PyPy distributions for macOS, Linux, and Windows.` in `Installing a Python version` section, what do you think?

**Review by zanieb — APPROVED** [review]
Thank you! Nice work.

**Pull request merged**